### ISSUE TITLE
Consistency between LDAP filter parameters in configuration

### ISSUE THREAD

**Issue by Erdou** [apereo/cas#2090]
Parameters in the configuration file are declared in different ways.
# Current state
### LDAP, everything except the LDAP attribute repository

```
# cas.authn.ldap[0].userFilter=cn={user}
# cas.authn.spnego.ldap.userFilter=cn={user}
# cas.monitor.ldap.userFilter=cn={user}
# cas.acceptableUsagePolicy.ldap.userFilter=cn={user}
# cas.mgmt.ldapAuthz.searchFilter=cn={user}
# cas.mgmt.ldapAuthz.userFilter=cn={user}
# cas.authn.pm.ldap.userFilter=cn={user}
```

Where user is obviously the username of the logged user.
### LDAP attribute repository

`# cas.authn.attributeRepository.ldap.userFilter=cn={0}`

First inconsistency here, with _0 = username_
### SPNego LDAP

`# cas.authn.spnego.ldap.searchFilter=host={0}`

This one is not a username but it still doesn't follow the named parameters logic.
### JDBC attribute repository

`# cas.authn.attributeRepository.jdbc.sql=SELECT * FROM table WHERE {0}`

This one is even more different, basically {0} will be replaced by "_usernameField_ = username" where usernameField comes from:
`# cas.authn.attributeRepository.jdbc.username=uid`
### JDBC authentication

`# cas.authn.jdbc.query[0].sql=SELECT password FROM table WHERE name=?`

Here it's a different type (SQL) of parameter which is used.
# Suggestions
- Keep all the named **{user}** parameters
- Change `# cas.authn.attributeRepository.ldap.userFilter=cn={0}` with `# cas.authn.attributeRepository.ldap.userFilter=cn={user}`
- Change `# cas.authn.spnego.ldap.searchFilter=host={0}` with `# cas.authn.spnego.ldap.searchFilter=host={host}`
- Change `# cas.authn.jdbc.query[0].sql=SELECT password FROM table WHERE name=?` with # `cas.authn.jdbc.query[0].sql=SELECT password FROM table WHERE name={user}`
- Change `# cas.authn.attributeRepository.jdbc.sql=SELECT * FROM table WHERE {0}` with `# cas.authn.attributeRepository.jdbc.sql=SELECT * FROM table WHERE name = {user}`

Note that for this last point, besides being consistent, it also gives more flexibility to the developer for building his query (ex. with a LIKE, even if that should never happen).

---
- Version: 5.0.0-RC4-SNAPSHOT

**Comment by mmoayyed** [comment]
Excellent analysis! Thanks.

**Comment by hdeadman** [comment]
I ran into this after copying userFilter: sAMAccountName={user} from cas.authn.ldap[0] to cas.authn.attributeRepository.ldap. It took a few times through the debugger before I saw what was wrong with the ldaptive SearchFilter. It would be helpful if person-directory or ldaptive would log the fact that the parameter is wrong because it is just running the query and returning no results. 

This code from LdaptivePersonAttributeDao.getPeopleForQuery() runs without complaining about what is a bad SearchFilter. 
```
            final Response<SearchResult> response;
            try {
                response = new SearchOperation(connection).execute(createRequest(filter));
            } catch (final LdapException e) {
                throw new RuntimeException("Failed executing LDAP query " + filter, e);
            }
            final SearchResult result = response.getResult();
```

**Comment by mmoayyed** [comment]
These two are now massaged to handle `{user}`:

```
# cas.authn.attributeRepository.ldap.userFilter=cn={user}
# cas.authn.spnego.ldap.searchFilter=host={host}
```

They also should still support for BC:

```
# cas.authn.attributeRepository.ldap.userFilter=cn={0}
# cas.authn.spnego.ldap.searchFilter=host={0}
```

JDBC changes will be done in later phases. While the above should be available in 5.1 SNAPSHOTs, 5.0.2 should be able to conceptually use the same patch by simply bumping person directory to the latest 1.8 release.

**Comment by mmoayyed** [comment]
JDBC changes are done.
